pkgs/test/test/runner/compiler_runtime_matrix_test.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+@TestOn('vm')
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:test_api/backend.dart'; // ignore: deprecated_member_use
+import 'package:test_descriptor/test_descriptor.dart' as d;
+
+import '../io.dart';
+
+void main() {
+  setUpAll(() async {
+    await precompileTestExecutable();
+  });
+
+  for (var runtime in Runtime.builtIn) {
+    // Ignore the platforms we can't run on this OS.
+    if (runtime == Runtime.internetExplorer && !Platform.isWindows ||
+        runtime == Runtime.safari && !Platform.isMacOS) {
+      continue;
+    }
+    group('--runtime ${runtime.identifier}', () {
+      for (var compiler in runtime.supportedCompilers) {
+        group('--compiler ${compiler.identifier}', () {
```

Comment:
[optional] I'm not sure if the second level of `group` is adding value.

Could save 2 spaces of indentation in the tests.

```suggestion
    for (var compiler in runtime.supportedCompilers) {
      group('--runtime ${runtime.identifier} --compiler ${compiler.identifier}', () {
```